junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/FilePosition.java

```diff
@@ -29,17 +29,21 @@
 
 	private static final long serialVersionUID = 1L;
 
-	private final int line;
-	private final int column;
-
 	/**
 	 * Create a new {@code FilePosition} using the supplied {@code line} and
 	 * {@code column}.
 	 *
 	 * @param line the line (1-based)
 	 * @param column the column (1-based)
 	 */
-	public FilePosition(int line, int column) {
+	public static FilePosition from(int line, int column) {
```

Comment:
@sbrannen I wonder if this SFM would be better named something like `fromLineAndColumn(int, int)`, as looking at the usages of the SFM in the tests below, they confuse me I cannot tell what the two `int` parameters mean without looking at the javadoc.

Reply:
That's a valid point. However, if we make an exception here, do we also need to use more specific names in all `TestSource` implementations?

Reply:
I personally don't see a need to do the same for `TestSource` implementations, since they only accept one argument, and those arguments are self-descriptive AFAICT.

Reply:
My concern is when we e.g. have two consecutive `int` parameters, which (1) are easy to confuse with one another, and (2) don't have any inherent meaning to their type apart from being whole numbers.

Reply:
This may warrant using the Builder pattern instead of SFMs for `FilePosition` specifically, so rather than having the following:

```java
FilePosition f = FilePosition.from(10, 10);
```

we'd have this instead, which IMO is easier to understand.

```java
FilePosition f = FilePosition.builder()
    .line(10)
    .column(10)
    .build():
```

Reply:
This would allow us to have poor-man's named parameters a la Python.

Reply:
Looks a bit over-engineered for two ints, doesn't it?

Reply:
Yes, looking back over this, I agree with you that using Builder here would over-engineer things a bit.

I still advocate for at least an SFM like `fromLineAndColumn`, but I can see that this is starting to fall into bikeshedding territory, so if you and @sbrannen disagree with that name and prefer to stick with `from`, then I'll relent. :)